ansible_builder/main.py

```diff
@@ -145,62 +175,63 @@ def get_dependency(self, entry):
 
         return os.path.join(self.reference_path, req_file)
 
-    def collection_dependencies(self, dep_type='python'):
-        """Returns a list of files for the dependency type
-        These are the dependency files declared by collections which
-        the user definition listed in its Galaxy requirement file
-        in other words, this returns indirect dependencies of given type
-        """
-        ret = []
-        for path in self.manager.path_list():
-            CD = CollectionDefinition(path)
-            dep_file = CD.get_dependency(dep_type)
-            if not dep_file:
-                continue
-            namespace, name = CD.namespace_name()
-            ret.append(os.path.join(namespace, name, dep_file))
-        return ret
-
 
 class Containerfile:
     newline_char = '\n'
 
     def __init__(self, definition,
                  filename=constants.default_file,
                  build_context=constants.default_build_context,
-                 base_image=constants.default_base_image):
+                 base_image=constants.default_base_image,
+                 container_runtime=constants.default_container_runtime,
+                 tag=constants.default_tag):
 
         self.build_context = build_context
         os.makedirs(self.build_context, exist_ok=True)
         self.definition = definition
         self.path = os.path.join(self.build_context, filename)
         self.base_image = base_image
+        self.container_runtime = container_runtime
+        self.tag = tag
+        self.steps = []
 
-    def build_steps(self):
+    def prepare_galaxy_steps(self):
         self.steps = [
             "FROM {}".format(self.base_image),
             ""
         ]
-        requirements_path = self.definition.get_dependency('galaxy')
-        if requirements_path:
+        galaxy_requirements_path = self.definition.get_dependency('galaxy')
+        if galaxy_requirements_path:
             # TODO: what if build context file exists? https://github.com/ansible/ansible-builder/issues/20
-            shutil.copy(requirements_path, self.build_context)
+            dest = os.path.join(self.build_context, galaxy_requirements_path)
+            exists = os.path.exists(dest)
+            if not exists or not filecmp.cmp(galaxy_requirements_path, dest, shallow=False):
+                shutil.copy(galaxy_requirements_path, dest)
```

Comment:
For the example I had, this is giving me:

```
Traceback (most recent call last):
  File "/Users/alancoding/.virtualenvs/ansible-builder/bin/ansible-builder", line 33, in <module>
    sys.exit(load_entry_point('ansible-builder', 'console_scripts', 'ansible-builder')())
  File "/Users/alancoding/Documents/repos/ansible-builder/ansible_builder/cli.py", line 85, in run
    if action():
  File "/Users/alancoding/Documents/repos/ansible-builder/ansible_builder/main.py", line 48, in build
    self.containerfile.prepare_galaxy_steps()
  File "/Users/alancoding/Documents/repos/ansible-builder/ansible_builder/main.py", line 209, in prepare_galaxy_steps
    shutil.copy(galaxy_requirements_path, dest)
  File "/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/shutil.py", line 248, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/shutil.py", line 121, in copyfile
    with open(dst, 'wb') as fdst:
FileNotFoundError: [Errno 2] No such file or directory: 'examples/bc/examples/pytz/requirements.yml'
```

I think the usage is reasonable, and I'm sure we can get a unit test to cover this.